ehsm_kms_service/ehsm_kms_server.js

```diff
@@ -43,7 +49,24 @@ const _checkSign = function (req,res,next){
     res.send(result(400,'Missing required parameters'));
     return;
   }
-  // todo sign 
+  let test_app_key = '202112345678';
+
+  let str = '';
+  let sign_parmas = {appid, nonce, timestamp} 
+  for(var k in sign_parmas){
+    if(!str) {
+      str += k + '=' + sign_parmas[k]
+    } else{
+      str += '&' + k + '=' + sign_parmas[k]
+    }
+  }
+  str += '&app_key=' + test_app_key;
+  let local_sign = crypto.createHmac('sha256',test_app_key).update(str, 'utf8').digest('base64');
+  if(sign != local_sign) {
```

Comment:
BTW, are sign and local_sign in lower case or upper case?

Reply:
for the parameter sign, suggest to do it in next commit.